tests/benchmark/array_access.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2023 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+const size = 1e4;
+const arr = new Array<number>(size);
+const expect = 49999995000000;
+let res = 0;
+
+export function main() {
+    for (let i = 0, j = 0; i < 1e7; i++, j++) {
+        arr[j] = i;
```

Comment:
Can we add a test to use `i32` type as the index type?

Reply:
Updated the result